core/src/main/kotlin/org/evomaster/core/problem/external/service/ExternalServices.kt

```diff
@@ -1,26 +1,169 @@
+     * used for external service.
+     */
+    private fun getNextAvailableAddress() : String {
+        val nextAddress: String = nextIPAddress(lastIPAddress)
+
+        if (isAddressAvailable(nextAddress, WIREMOCK_PORT)) {
+            return nextAddress
+        } else {
+            throw Exception(nextAddress.plus(" is not available for use"))
+        }
+    }
+
+    /**
+     * Returns the default IP address for external service initialisation.
+     * If the respective address and port is not available it'll throw an
+     * exception.
+     */
+    private fun getDefaultAddress() : String {
+        if (isAddressAvailable("127.0.0.10", WIREMOCK_PORT)) {
+            return "127.0.0.10"
+        } else {
+            throw Exception("Default loopback address is unavailable for binding")
+        }
+    }
+
+    /**
+     * Will generate random IP address within the loopback range
+     * while checking the availability. If not available will
+     * generate a new one.
+     */
+    private fun generateRandomAvailableAddress() : String {
+        val ip = generateRandomIPAddress()
+        if (isAddressAvailable(ip, WIREMOCK_PORT)) {
+            return ip
+        }
+        return generateRandomAvailableAddress()
+    }
+
     fun getExternalServices() : List<ExternalServiceInfo> {
         return externalServiceInfos
     }
+
+    /**
+     * Default IP address will be a randomly generated IP
+     */
+    private fun getIP() : String {
+        val ip: String
+        when (config.externalServiceIPSelectionStrategy) {
+            // Although the default address will be a random, this
+            // option allows selecting explicitly
+            EMConfig.ExternalServiceIPSelectionStrategy.RANDOM -> {
+                ip = if (externalServiceInfos.size > 0) {
+                    getNextAvailableAddress()
+                } else {
+                    generateRandomAvailableAddress()
+                }
+            }
+            EMConfig.ExternalServiceIPSelectionStrategy.USER -> {
+                ip = if (externalServiceInfos.size > 0) {
+                    getNextAvailableAddress()
+                } else {
+                    config.externalServiceIP
+                }
+            }
+            else -> {
+                ip = if (externalServiceInfos.size > 0) {
+                    getNextAvailableAddress()
+                } else {
+                    generateRandomAvailableAddress()
+                }
+            }
+        }
+        return ip
+    }
+    /**
+     * Will initialise WireMock instance on a given IP address for a given port.
+     */
+    private fun initWireMockServer(address: String): WireMockServer {
+        val wm = WireMockServer(
+            WireMockConfiguration()
+                .bindAddress(address)
+                .port(WIREMOCK_PORT)
+                .extensions(ResponseTemplateTransformer(false)))
+        wm.start()
```

Comment:
as we start servers, we ll need a method to shut them all down (eg, to be called at end of the search)